mlflow/store/file_store.py

```diff
@@ -138,7 +139,18 @@ def list_experiments(self, view_type=ViewType.ACTIVE_ONLY):
             rsl += self._get_active_experiments(full_path=False)
         if view_type == ViewType.DELETED_ONLY or view_type == ViewType.ALL:
             rsl += self._get_deleted_experiments(full_path=False)
-        return [self._get_experiment(exp_id, view_type) for exp_id in rsl]
+        experiments = []
+        for exp_id in rsl:
+            try:
+                experiment = self._get_experiment(exp_id, view_type)
+                experiments.append(experiment)
+            except ResourceNotFoundException as rnfe:
+                # Trap malformed experiments and log warnings.
+                logging.warning("Malformed experiment '%s'. Detailed error %s", exp_id, str(rnfe))
```

Comment:
I think we should reraise the exception here. A client attempting to list experiments from a remote tracking server would benefit from the detailed error output that will only be propagated if the exception is reraised.